manifests/resource/location.pp

```diff
@@ -407,8 +407,12 @@
 
   $config_file = "${server_dir}/${server_sanitized}.conf"
 
-  $location_sanitized_tmp = regsubst($location, '\/', '_', 'G')
-  $location_sanitized = regsubst($location_sanitized_tmp, '\\\\', '_', 'G')
+  if !($location =~ /^~/) {
+    $location_sanitized_tmp = regsubst($location, '\/', '_', 'G')
+    $location_sanitized = regsubst($location_sanitized_tmp, '\\\\', '_', 'G')
+  } else {
+    $location_sanitized = $location_sanitized_tmp
```

Comment:
Shouldn't the right side be $localization? The current bar there will be uninitialized at this point.